app/models/schema.rb

```diff
@@ -0,0 +1,20 @@
+# Describes the JSON schema for a KimonoLabs endpoint as a series of effective, versioned structures.
+class Schema < ActiveRecord::Base
+  validates :structure, presence: true
+
+  belongs_to :endpoint, inverse_of: :schemas
+
+  # Initialize effective range
+  after_initialize :default_values
+  before_save :default_values
+
+  private
+
+  # Ideally we would do this with a database default, but ActiveRecord doesn't like that:
+
+  # ArgumentError: bad value for range
+  # from ... gems/activerecord-4.2.0/lib/active_record/connection_adapters/postgresql/oid/range.rb:39:in `initialize'
+  def default_values
+    self.effective ||= DateTime.new(1, 1, 1)..DateTime::Infinity.new
+  end
```

Comment:
The idea here is to leverage [Ruby range types using ActiveRecord and Postgres](http://edgeguides.rubyonrails.org/active_record_postgresql.html#range-types) so I can say things like `schema.effective.include?(Date.today)`, and leverage the other range operators.  I've dealt with periodic data like this in other projects and date ranges are fantastic for them.

That said, I'm getting the sense that [this isn't as done](https://github.com/rails/rails/issues/17288) as [I initially thought](https://github.com/rails/rails/issues/14010).  Going to play with it some more and see.